### DISCUSSION TITLE
How to delete all rows of a table using the public API?

### DISCUSSION THREAD

**Discussion by Hedronmx** [Budibase/budibase#9974] [Help] [answered]
Hello! I want to retrieve all rows of a table and delete them all every two weeks. But i dont know how to get all the rows of a table? Whats the api request like?

Thank you!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Hedronmx, 

You can find alot of that information here in the docs. Hopefully this is helpful to you.
https://docs.budibase.com/reference/get_tables-tableid

**Reply by ConorWebb96** [reply]
Follow up on the previous response, 

You will likely have to use the search request on the public API: 
https://docs.budibase.com/reference/post_tables-search

**An example of retreving the data through the public API**
![retrievedData](https://user-images.githubusercontent.com/126772285/224287938-c24f5b05-1296-43d0-9943-583074ac0fa1.png)

Increase the pagination limit to 1000 to get the max amount of results. If its more than 1000 you will get a bookmark with each response, allowing you to retrieve the next batch of data.

From here you would need to loop through each batch's data and run the delete row request on each row.
https://docs.budibase.com/reference/delete_tables-tableid-rows-rowid

Hopefully this helps.
